src/v/raft/group_configuration.cc

```diff
@@ -161,34 +272,24 @@ group_configuration::group_configuration(
   std::optional<group_nodes> old,
   version_t v)
   : _version(v)
-  , _brokers(std::move(brokers))
+  // do not set brokers for v_5 and next versions
+  , _brokers(_version < v_5 ? std::move(brokers) : std::vector<model::broker>{})
```

Comment:
nit: should probably add an assert that brokers is empty for v5 instead of silently ignoring the passed value